kernel-spark/src/main/java/io/delta/kernel/spark/utils/ExpressionUtils.java

```diff
@@ -0,0 +1,259 @@
+      return convertValueToKernelLiteral(f.value())
+          .map(l -> new Predicate(">", column(f.attribute()), l));
+    }
+    if (filter instanceof GreaterThanOrEqual) {
+      GreaterThanOrEqual f = (GreaterThanOrEqual) filter;
+      return convertValueToKernelLiteral(f.value())
+          .map(l -> new Predicate(">=", column(f.attribute()), l));
+    }
+    if (filter instanceof LessThan) {
+      LessThan f = (LessThan) filter;
+      return convertValueToKernelLiteral(f.value())
+          .map(l -> new Predicate("<", column(f.attribute()), l));
+    }
+    if (filter instanceof LessThanOrEqual) {
+      LessThanOrEqual f = (LessThanOrEqual) filter;
+      return convertValueToKernelLiteral(f.value())
+          .map(l -> new Predicate("<=", column(f.attribute()), l));
+    }
+    if (filter instanceof IsNull) {
+      IsNull f = (IsNull) filter;
+      return Optional.of(new Predicate("IS_NULL", column(f.attribute())));
+    }
+    if (filter instanceof IsNotNull) {
+      IsNotNull f = (IsNotNull) filter;
+      return Optional.of(new Predicate("IS_NOT_NULL", column(f.attribute())));
+    }
+    if (filter instanceof org.apache.spark.sql.sources.And) {
+      org.apache.spark.sql.sources.And f = (org.apache.spark.sql.sources.And) filter;
+      Optional<Predicate> left = convertSparkFilterToKernelPredicate(f.left(), canPartialPushDown);
+      Optional<Predicate> right =
+          convertSparkFilterToKernelPredicate(f.right(), canPartialPushDown);
+      if (left.isPresent() && right.isPresent()) {
+        return Optional.of(new And(left.get(), right.get()));
+      }
+      if (canPartialPushDown && left.isPresent()) {
+        return left;
+      }
+      if (canPartialPushDown && right.isPresent()) {
+        return right;
+      }
+      return Optional.empty();
+    }
+    if (filter instanceof org.apache.spark.sql.sources.Or) {
+      org.apache.spark.sql.sources.Or f = (org.apache.spark.sql.sources.Or) filter;
+      Optional<Predicate> left = convertSparkFilterToKernelPredicate(f.left(), canPartialPushDown);
+      Optional<Predicate> right =
+          convertSparkFilterToKernelPredicate(f.right(), canPartialPushDown);
+      // OR requires both operands to be convertible for correctness
+      if (!left.isPresent() || !right.isPresent()) {
+        return Optional.empty();
+      }
+      return Optional.of(new Or(left.get(), right.get()));
+    }
+    if (filter instanceof Not) {
+      Not f = (Not) filter;
+      // NOT disables partial pushdown for semantic correctness
+      // NOT (A AND B) cannot become NOT A - they have different meanings
+      Optional<Predicate> child = convertSparkFilterToKernelPredicate(f.child(), false);
+      return child.map(c -> new Predicate("NOT", c));
+    }
+
+    return Optional.empty();
+  }
+
+  /**
+   * Creates a Delta Kernel Column from a Spark SQL column attribute name.
+   *
+   * <p>This method handles nested column references (e.g., "user.profile.name") by parsing the
```

Comment:
what happens if a column has a dot inside of it?

Reply:
Add a comment, if column name has doc, it should be passed as "`user.profile.name`"